client/signup/steps/about/index.jsx

```diff
@@ -172,19 +169,10 @@ class AboutStep extends Component {
 	};
 
 	getSuggestions() {
-		const query = this.state.query && escapeRegExp( this.state.query ).toLowerCase();
-		const regex = new RegExp( query, 'i' );
-
-		// Prioritize suggestions starting with query input first
-		const sortFunction = ( a, b ) =>
-			abtest( 'aboutSuggestionMatches' ) === 'enhancedSort'
-				? a.localeCompare( b ) -
-				  2 * ( b.toLowerCase().indexOf( query ) - a.toLowerCase().indexOf( query ) )
-				: a.localeCompare( b );
-
+		const { query } = this.state;
 		return Object.values( hints )
-			.filter( hint => query && hint.match( regex ) )
-			.sort( sortFunction )
+			.filter( hint => query && hint.match( new RegExp( escapeRegExp( query ), 'i' ) ) )
```

Comment:
I'm used to seeing `escapeRegExp()` used when mixing a RegExp pattern we make with user-supplied data…

```js
const wordList = userInput.text.split( ' ' ).map( escapeRegExp );
const pattern = wordList.join( '|' );
pattern.test( stringContainsOneOrMoreWordsFromTheList );
```

it's a bit surprising to find it being passed in with no other pattern, maybe other than the case-insensitivity flag.
would we have a reason to avoid the more basic string searching?

```js
const { query: rawQuery } = this.state;

if ( ! rawQuery ) {
	return [];
}

const query = rawQuery.toLocaleLowerCase();

return Object.values( hints )
	.filter( hint => hint.toLocaleLowerCase().includes( query ) )
	.sort( ( a, b ) => a.localeCompare( b ) )
	.map( hint => ( { label: hint } ) );
```

Reply:
String searching sounds 👍!